el-patch-template.el

```diff
@@ -0,0 +1,513 @@
+;;; el-patch-template.el --- -*- lexical-binding: t -*-
+
+
+;; Author: Al Haji-Ali <abdo.haji.ali@gmail.com>
+;; Created: 1 March 2021
+;; Homepage: http://www.randomoid.com
+;; Keywords: extensions
+;; Package-Requires: ((emacs "25"))
+;; SPDX-License-Identifier: MIT
+;; Version: 0.0.1
+
+;;; Commentary:
+;; `el-patch-template' is an extension of `el-patch' that allows one
+;; to specifiy a patch without providing the complete source code of
+;; the patched form.
+;;
+;; Example usage:
+;;
+;; (el-patch-template (defun TeX-update-style)
+;;   (el-patch-concat
+;;     "Run style specific hooks"
+;;     (el-patch-add
+;;       ", silently,")
+;;     " for the current document.
+;;
+;; Only do this if it has not been done before, or if optional argument
+;; FORCE is not nil.")
+;;   (el-patch-remove
+;;     (... "Applying style hooks..."))
+;;   (el-patch-remove
+;;     (... "Applying style hooks...done")))
+
+(defun el-patch-template--process-el-patch
+    (form template &optional match next-step-fn table)
+  "Processes an el-patch statement. Arguments are the same as
+`el-patch-template--match'. Assume thats template is a cons whose
```

Comment:
The style guide for Elisp docstrings says to put argument names in all uppercase, and to write in the imperative verb tense ("Do X" rather than "Does X").

```suggestion
  "Process an el-patch statement. Arguments are the same as
`el-patch-template--match'. Assumes that TEMPLATE is a cons whose
```

Similar comments apply to the other docstrings, I think.